src/test/java/com/base2services/jenkins/GitHubTriggerProcessorTest.java

```diff
@@ -72,7 +72,7 @@ public void shouldNotTriggerWithAFork() throws Exception {
     public void shouldNotTriggerWithBadPayload() throws Exception {
         payload = "{'noRepoInfo': {'empty': 'https://github.com/foo/bar'}}".replace("'", "\"");
         GitHubTriggerProcessor gtp = new GitHubTriggerProcessor();
-        gtp.processGitHubPayload(payload, sbt.getClass());
+        gtp.trigger(payload);
```

Comment:
Reviewer note:
This worked before because it was relying on the logic in the extractJsonFromPayload function that would have it return null if there was no 'repository' object in the payload. The new code changes will catch this case earlier, in the trigger function.